doc/man7/RAND_DRBG.pod

```diff
@@ -0,0 +1,315 @@
+=pod
+
+=head1 NAME
+
+RAND_DRBG - the deterministic random bit generator
+
+=head1 SYNOPSIS
+
+ #include <openssl/rand_drbg.h>
+
+=head1 DESCRIPTION
+
+The default OpenSSL RAND method is based on the RAND_DRBG class,
+which implements a deterministic random bit generator (DRBG).
+A DRBG is a certain type of cryptographically-secure pseudo-random
+number generator (CSPRNG), which is described in
+[NIST SP 800-90A Rev. 1].
+
+While the RAND API is the 'frontend' which is intended to be used by
+application developers for obtaining random bytes, the RAND_DRBG API
+serves as the 'backend', connecting the former with the operating
+systems's entropy sources and providing access to the DRBG's
+configuration parameters.
+
+=head2 Disclaimer
+
+Unless you have very specific requirements for your random generator,
+it is in general not necessary to utilize the RAND_DRBG API directly.
+The usual way to obtain random bytes is to use L<RAND_bytes(3)> or
+L<RAND_priv_bytes(3)>, see also L<RAND(7)>.
+
+=head2 Typical Use Cases
+
+Typical examples for such special use cases are the following:
+
+=over 2
+
+=item *
+
+You want to use your own private DRBG instances, similar to how it
+is currently done in the ssl library.
+Multiple DRBG instances which are accessed only by a single thread provide
+additional security (because their internal states are independent) and
+better scalability in multithreaded applications (because they don't need
+to be locked).
+
+=item *
+
+You need to integrate a previously unsupported entropy source.
+
+=item *
+
+You need to change the default settings of the standard OpenSSL RAND
+implementation to meet specific requirements.
+
+=back
+
+
+=head1 CHAINING
+
+A DRBG instance can be used as the entropy source of another DRBG instance,
+provided it has itself access to a valid entropy source.
+
+The DRBG instance which acts as entropy source is called the I<parent> DRBG,
+the other instance the I<child> DRBG.
+
+This is called chaining. It is possible create chains of more than two DRBG
+in a row.
+
+=head1 THE THREE SHARED DRBG INSTANCES
+
+By default, the functions L<RAND_bytes(3)> and L<RAND_priv_bytes(3)> use
+global DRBG instances, which are accessed concurrently by all threads.
+The necessary locking is done automatically by the RAND methods, e.g.,
+by L<RAND_bytes(3)> and L<RAND_priv_bytes(3)>.
+
+Currently, there are three DRBG instances,
+the <master>, <public>, and <private> DRBG.
```

Comment:
I'm wondering if we should document this or not. Your other pull request doesn't actually seem to make those 3 available, which might just be an oversight. And as long as they are not available, I see little point in making public documentation for it and prefer to keep it internal documentation.

Reply:
That was indeed an oversight, thanks. Documentation for `RAND_DRBG_get0_*` has been added.